File: Java OOP-Feb2020/src/i_polymorphism_Lab/animals/Animal.java
package i_polymorphism_Lab.animals;

public abstract class Animal {
    private String name;
    private String favouriteFood;

    public Animal(String name, String favouriteFood) {
        this.name = name;
        this.favouriteFood = favouriteFood;
    }

    public String explainSelf() {
        return String.format("I am %s and my favourite food is %s", this.name, this.favouriteFood);
    }

    protected String getName() {
        return name;
    }

    protected String getFavouriteFood() {
        return favouriteFood;
    }
}

File: Programming-for-QA-sep2023/20.Encapsulation-Inheritance-Abstraction-Polymorphism/07.Animals/Models/Animal.cs
namespace Animals.Models;

public abstract class Animal
{
    private string _name;
    private string _favoriteFood;

    //public string Name { get { return _name; } }
    //public string FavoriteFood {  get { return _favoriteFood; } }

    protected Animal(string name, string favoriteFood)
    {
        this._name = name;
        this._favoriteFood = favoriteFood;
    }

    public virtual string ExplainSelf()
    {
        return $"I am {this._name} and my fovourite food is {this._favoriteFood}";
    }
}
